zebedee-reader/src/main/java/com/github/onsdigital/zebedee/reader/api/endpoint/Generator.java

```diff
@@ -65,14 +85,18 @@ public void get(HttpServletRequest request, HttpServletResponse response) throws
     private Resource toResource(Content content, String format) throws IOException, BadRequestException {
         if (content instanceof Chart) {
             // If page is a chart write the chart spreadsheet requested to response
-            return new DataGenerator().generateData((Chart) content, format);
+            return DATA_GENERATOR.generateData((Chart) content, format);
         } else if (content instanceof TimeSeries) {
             // If page then write the timeseries spreadsheet requested to response
-            return new DataGenerator().generateData((TimeSeries) content, format);
+            return DATA_GENERATOR.generateData((TimeSeries) content, format);
         } else if (content instanceof Series) {
-            return new DataGenerator().generateData((Series) content, format);
+            return DATA_GENERATOR.generateData((Series) content, format);
```

Comment:
There a small fix you can do in the GeneratorData. Instead of writing to disk, use
```
OutputStream os =  new ByteArrayOutputStream()
```
Make sure the stream gets closed.